python/lsst/meas/extensions/psfex/psfexPsfDeterminer.py

```diff
@@ -367,8 +415,18 @@ def determinePsf(self, exposure, psfCandidateList, metadata=None, flagKey=None):
         # the PSF kernel and checking for an InvalidParameterError.
         try:
             _ = psf.getKernel(psf.getAveragePosition())
-        except pexExcept.InvalidParameterError:
-            raise RuntimeError("Failed to determine psfex psf: too few good stars.")
+        except pexExcept.InvalidParameterError as e:
+            ndim = int(str(e).split('saw ')[1].split()[0])
```

Comment:
I don't like trying to parse an exception string: that's typically an implementation detail and could change out from under us. Is there any other way to get information back from psfex?

Reply:
I agree, I don't like it either. I tried everything but couldn't find another way. Any suggestions on tweaking the other codes to pass the info more robustly?

Reply:
Where in the psfex code is this exception raised? Looks like it's in our own C++ wrapper, `PsfexPsf.cc:233`, I think. So instead of this `try:except` block, is there a getter (or could we add one) to the `PsfexPsf` class that would let us get `ndim`, and then we can test it directly here?

Reply:
Correct, I was just trying to keep it python-only at first, but editing our own C++ wrapper seems like the right move for this situation. I don't see a `ndim` getter in `PsfexPsf`, but I can definitely add one!

Reply:
Done!